test/11_array.jl

```diff
@@ -132,6 +132,38 @@ for (A, V, M) in ((NominalArray, NominalVector, NominalMatrix),
         @test x[3] === x.pool.valindex[1]
         @test levels(x) == ["e", "a", "b", "c"]
 
+        x = V{String, R}(a)
+        push!(x, "a")
+        @test length(x) == 4
+        @test String(x[end]) == "a"
+
+        push!(x, "zz")
+        @test length(x) == 5
+        @test String(x[end]) == "zz"
+
+        push!(x, x[1])
+        @test length(x) == 6
+        @test x[1] == x[end]
+
+        y = V{String, R}(a)
+        append!(x, y)
+        @test length(x) == 9
+
+        a = ["a","b","c"]
+        x = V{String, R}(a)
+        a = ["z","y","x"]
+        y = V{String, R}(a)
+        append!(x, y)
+        @test length(x) == 6
+        @test String(x[1]) == "a"
+        @test String(x[2]) == "b"
+        @test String(x[3]) == "c"
+        @test String(x[4]) == "z"
+        @test String(x[5]) == "y"
+        @test String(x[6]) == "x"
+
+        empty!(x)
+        @test length(x) == 0
```

Comment:
Could you also test `levels` here and for other functions? A particularly important case is when `x` and `y` share some levels, but not in the same order. Should be easy to add to your existing tests.